### ISSUE TITLE
Improve error messages when transformer plugins fail to load

### ISSUE THREAD

**Issue by DartBot** [dart-lang/pub#896]
<a href="https://github.com/yjbanov"><img src="https://avatars.githubusercontent.com/u/211513?v=3" align="left" width="96" height="96"hspace="10"></img></a> **Issue by [yjbanov](https://github.com/yjbanov)**
_Originally opened as dart-lang/sdk#17200_

---

**What steps will reproduce the problem?**
1. Create two projects A and B
2. B depends on A
2. Create a transformer in project A
2. Add it to B's pubspec.yaml
3. 'pub serve' project B

**What is the expected output? What do you see instead?**

Expecting 'pub serve' not to crash. Instead, it crashes (stderr output from pub --trace 'serve' attached).

**What version of the product are you using? On what operating system?**

barback 0.11.1
Tried two Dart VM versions:
1.3.0-dev.0.0 (Wed Feb 26 07:39:42 2014) on &quot;macos_x64&quot;
1.2.0-dev.5.15 (Mon Feb 24 02:12:58 2014) on &quot;macos_x64&quot;

---

**Attachment:**
[barback-sterr.log](https://storage.googleapis.com/google-code-attachments/dart/issue-17200/comment-0/barback-sterr.log) (146.37 KB)

**Comment by DartBot** [comment]
<a href="https://github.com/iposva-google"><img src="https://avatars.githubusercontent.com/u/5449880?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [iposva-google](https://github.com/iposva-google)**

----

*Added [Pkg-Barback](../labels/Pkg-Barback), [Area-Pkg](../labels/Area-Pkg), [Triaged](../labels/Triaged) labels.*

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

I can't reproduce this locally. Can you confirm this is still happening on the latest bleeding edge pub version, and if so can you provide a tarball of an app where this happens?

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

*Added [CannotReproduce](../labels/CannotReproduce) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/yjbanov"><img src="https://avatars.githubusercontent.com/u/211513?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [yjbanov](https://github.com/yjbanov)**

----

So, it turns out I was incorrectly referring to the transformer. I needed to specify full path to the file where the transformer is defined, but instead I just used the package name. Bob N. says for that to work I have to name the file transformer.dart. I'll try that later.

However, that's not an excuse to produce illegible errors. It should clearly say "your transformer cannot be found" or sth like that, preferably w/o stack traces. So, I tried again. I cannot reproduce the exact same error, but I get a different error, this time even less legible that the last. Attaching both the stderr and the archived sample projects.

Dart VM version: 1.3.0-dev.3.2 (Mon Mar 10 11:06:13 2014) on "macos_x64"
Pub 1.3.0-dev.3.2

______
**Attachments:**
[stderr.txt](https://storage.googleapis.com/google-code-attachments/dart/issue-17200/comment-4/stderr.txt) (929 Bytes)
[sample-projects.zip](https://storage.googleapis.com/google-code-attachments/dart/issue-17200/comment-4/sample-projects.zip) (4.64 KB)

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

It would be nice to improve these error messages, but currently the IsolateSpawnException API doesn't expose any information we can use to do so. I've filed issue dart-lang/sdk#17492 to track that.
______
*Removed [Pkg-Barback](../labels/Pkg-Barback), [Area-Pkg](../labels/Area-Pkg) labels.*
*Added [Area-Pub](../labels/Area-Pub), [Triaged](../labels/Triaged) labels.*
*Marked this as being blocked by dart-lang/sdk#17492.*
*Changed the title to: "Improve error messages when transformer plugins fail to load".*

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

Oops, I mean issue dart-lang/sdk#17501.
______
*Marked this as being blocked by dart-lang/sdk#17501.*
*Unmarked this as being blocked by dart-lang/sdk#17492.*

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

Issue dart-lang/pub#772 has been merged into this issue.

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

Issue dart-lang/sdk#19961 has been merged into this issue.

**Comment by matanlurey** [comment]
We are no longer supporting transformers or `pub serve/build` going forward (this tooling has moved elsewhere into the https://github.com/dart-lang/build repository and https://github.com/dart-lang/webdev CLI) so a fix for this is not planned. 

We'll be updating the tooling section of the Dart website in the coming weeks with more formal documentation on how to get started.
